include/psa/crypto_builtin_composites.h

```diff
@@ -112,4 +112,59 @@ typedef struct
 
 #define MBEDTLS_PSA_AEAD_OPERATION_INIT {0, 0, 0, 0, {0}}
 
+#include "mbedtls/ecdsa.h"
+
+/* Context structure for the Mbed TLS interruptible sign hash implementation. */
+typedef struct
+{
+
+#if defined(MBEDTLS_PSA_BUILTIN_ALG_ECDSA) || \
+ defined(MBEDTLS_PSA_BUILTIN_ALG_DETERMINISTIC_ECDSA)
+    mbedtls_ecdsa_context * MBEDTLS_PRIVATE( ctx );
+#if defined( MBEDTLS_ECP_RESTARTABLE )
```

Comment:
Do you mean wrapping the whole struct contents with #if defined(  MBEDTLS_ECP_RESTARTABLE ), or making the struct not exist if `MBEDTLS_ECP_RESTARTABLE` is not defined, or something else?

Reply:
I was thinking wrapping the whole struct contents. Basically my reasoning is that if we have ECDSA (one of the flavours) but not RESTARTABLE, then does it really make sense to have everything but the restartable context here, as we won't be able to actually do any restartable operation anyway? Or am I missing something?

Also, looking at other places, `(BUILTIN_ECDSA || BUILTIN_DETERMINSTIC_ECDSA) && ECP_RESTARTABLE` seem quite common so it would seem sensible to have the same here.